toolchain/lower/function_context.cpp

```diff
@@ -121,6 +121,15 @@ auto FunctionContext::LowerInst(SemIR::InstId inst_id) -> void {
   }
 }
 
+auto FunctionContext::Finish() -> void {
+  // Move allocas into the entry block.
+  auto& entry_block = function_->getEntryBlock();
+  auto insert_point = entry_block.getFirstNonPHIOrDbgOrAlloca();
```

Comment:
Hmm, shouldn't we ensure that the entry block isn't branched to? And given that, there shouldn't be PHIs or other instructions canonically?

I guess you want to skip over allocas placed while emitting the entry block itself though? That makes sense, but maybe the suggestion I make in the other inline comment can avoid needing this...